server/server.go

```diff
@@ -2307,6 +2331,73 @@ func (s *Server) v1StatusGet(w http.ResponseWriter, r *http.Request) {
 	writer.JSON(w, http.StatusOK, resp, pretty)
 }
 
+func (s *Server) validateInput(input interface{}) []types.BadRequestErr {
+
+	info, err := os.Stat(s.schemaPath)
+	if err != nil {
+		return []types.BadRequestErr{types.BadRequestErr(err.Error())}
+	}
+
+	if !info.IsDir() {
+		bs, err := os.ReadFile(s.schemaPath)
+		if err != nil {
+			return []types.BadRequestErr{types.BadRequestErr(err.Error())}
+		}
+
+		result, err := gojsonschema.Validate(gojsonschema.NewBytesLoader(bs), gojsonschema.NewGoLoader(input))
+		if err != nil {
+			return []types.BadRequestErr{types.BadRequestErr(err.Error())}
+		}
+
+		if result.Valid() {
+			return nil
+		}
+
+		errs := []types.BadRequestErr{}
+		for _, err := range result.Errors() {
+			errs = append(errs, types.BadRequestErr(err.String()))
+		}
+		return errs
+	}
+
+	errs := []types.BadRequestErr{}
+	err = filepath.Walk(s.schemaPath,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			} else if info.IsDir() {
+				return nil
+			}
+
+			bs, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			result, err := gojsonschema.Validate(gojsonschema.NewBytesLoader(bs), gojsonschema.NewGoLoader(input))
```

Comment:
Can't json schemas have references to other schema files? If so, those references would be broken if a separate validation is done for each schema. The compiler has a `mergeSchemas` function, but I don't know if that's related.

Reply:
Thanks. I'll look into this.

Reply:
@johanfylling I think I remember seeing something like that in the spec. Related docs: https://json-schema.org/understanding-json-schema/structuring.html#json-pointer

Reply:
Updated to support schemas having references to other schemas. Tests pending.